docs/src/snap/howto/ceph.md

```diff
@@ -0,0 +1,306 @@
+# How to use ceph
+
+Distributed, redundant storage is a must-have when you want to develop reliable applications. Ceph is a storage solution which provides exactly that, and is built with distributed clusters in mind. In this tutorial, we'll be looking at how to integrate Canonical Kubernetes with a Ceph cluster. Specifically, by the end of this tutorial you'll have a Kubernetes pod with a mounted RBD-backed volume. This how to is adapted from [block-devices-and-kubernetes].
+
+## What you'll need
+
+This guide assumes the following:
+
+- You have root or sudo access to the machine
+- You have a bootstrapped Canonical Kubernetes cluster (see the
+  [getting-started-guide])
+- You have a running Ceph cluster
+
+## Create a Ceph storage pool
+
+Create a storage pool named "kubernetes" in the Ceph cluster.
+
+We will set the number of placement groups to 128 because the Ceph cluster of this demonstration will have less than 5 OSDs. (See [placement groups])
+
+```
+ceph osd pool create kubernetes 128
+```
+
+Initialize the pool as a Ceph block device (RBD) pool.
+
+```
+rbd pool init kubernetes
+```
+
+## Configure ceph-csi
+
+Ceph CSI is the CSI driver for Ceph. With Ceph CSI, Kubernetes will be able to accomplish tasks related to your Ceph cluster (like attaching volumes to workloads.)
+
+Create a user for Kubernetes and ceph-csi.
+
+```
+ceph auth get-or-create client.kubernetes mon 'profile rbd' osd 'profile rbd pool=kubernetes' mgr 'profile rbd pool=kubernetes'
+```
+
+```
+[client.kubernetes]
+	key = AQBh1TNmFYERJhAAf5yqP4Wnrb/u4yNGsBKZHA==
+```
+
+Note the generated key.
+
+## Generate ceph-csi-configmap.yaml
```

Comment:
I think this whole section need a bit more love.

Add a bit of introduction, why do we need this configmap?